core/invoker/src/main/scala/whisk/core/containerpool/kubernetes/KubernetesContainerFactory.scala

```diff
@@ -39,7 +39,7 @@ class KubernetesContainerFactory(label: String, config: WhiskConfig)(implicit ac
                                                                      logging: Logging)
     extends ContainerFactory {
 
-  implicit val kubernetes = new KubernetesClient()(ec)
+  implicit lazy val kubernetes: KubernetesApi = new KubernetesClient()(ec)
```

Comment:
Is there a reason to make this lazy? Lazys *are* fine, though if unnecessary worth not using since their initialization is sequential, which introduces locks and a small performance hit (should be entirely unnoticeable here, but I'd rather just double check "do we need a lazy?")

Reply:
The change to lazy val was so that I could override it easily to a `TestKubernetesClient` in `KubernetesContainerFactoryTest` - see line 55 of that file. Otherwise, `KubernetesContainerFactory` tried to construct a real Kube client, looking for Kube binaries and other initialization stuff that's not needed in the test.

I didn't know the right way to do this, and I've never used lazy vals in Scala before so let me know if there's a more appropriate way to accomplish the same thing.